src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,135 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int ARRAYS_ZERO_BORDER = 0;
+    private static final String INDEX_OUT_OF_BOUNDS = "Index out of bounds array size";
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        head = new Node<>(null, null, null);
+        tail = new Node<>(null, null, null);
+    }
+
     @Override
     public void add(T value) {
+        if (!addFirstNode(value)) {
+            tail.prev = tail.prev.next = new Node<>(tail.prev, value, tail);
+            size++;
```

Comment:
try to think of it this way: we can create a node, pointing straight to tail. and think about what can we do based on different size conditions?